### PR TITLE
Use http2-manager to communicate with remotes

### PR DESCRIPTION
**Pull request by akshaymankar**
## Checklist

 - [x] Add a new entry in an appropriate subdirectory of `changelog.d`
 - [x] Read and follow the [PR guidelines](https://docs.wire.com/developer/developer/pr-guidelines.html)

### CODE REVIEW

File: changelog.d/5-internal/reuse-http2 [hunk 1]
```diff
@@ -1 +1 @@
-Reuse HTTP2 connections from brig, galley, cargohold and federator
\ No newline at end of file
+Reuse HTTP2 connections from brig, galley, cargohold and federator (#3120, #3233)
```
Comment (akshaymankar) [inline changelog.d/5-internal/reuse-http2]: Note to reviewer, in the previous PR I made a grammatical mistake by saying "from ... and", I meant to say "
from ... to", but with this PR the mistake is useful.

File: services/federator/test/unit/Test/Federator/Monitor.hs [hunk 2]
```diff
@@ -119,7 +119,7 @@ withSilentMonitor reloads settings = do
   tlsVar <- liftIO $ newIORef (error "TLSSettings not updated before being read")
   void . ContT $
     bracket
-      (runSem (mkMonitor runSemE tlsVar settings))
+      (runSem (mkMonitor runSemE (atomicWriteIORef tlsVar) settings))
```
Comment (fisx) [inline services/federator/test/unit/Test/Federator/Monitor.hs:122]: this is not a tlsVar any more, no?

**Review by fisx — APPROVED** [review]
approved with two caveats.  also i haven't checked the tests, but i'm assuming there are integration tests covering these changes?

File: services/federator/test/unit/Test/Federator/Monitor.hs (hunk 2 above)
Comment (akshaymankar) [reply services/federator/test/unit/Test/Federator/Monitor.hs:122]: It is an `IORef` same as earlier, see 3 lines above.

**Pull request merged**